lib/hocon/impl/simple_config_object.rb

```diff
@@ -107,6 +256,153 @@ def merged_with_object(abstract_fallback)
     end
   end
 
+  def modify(modifier)
+    begin
+      modify_may_throw(modifier)
+    rescue Hocon::ConfigError => e
```

Comment:
Good question; I wasn't sure what to use because I don't think there is any sane semantic mapping from Java's `RuntimeException` and, well, Ruby.  So I figured we'd just iterate on this when/if a test turns up something actionable?